src/shell/components/CascadingMenuItem/index.tsx

```diff
@@ -0,0 +1,61 @@
+import React, { FC, useState } from "react";
+import {
+  MenuItem,
+  MenuItemProps,
+  Popper,
+  PopperProps,
+  Paper,
+  PaperProps,
+} from "@mui/material";
+import { theme } from "@zesty-io/material";
+
+type CascadingMenuItemProps = MenuItemProps & {
+  MenuItemComponent: React.ReactNode;
+  children: React.ReactNode;
+  PopperProps?: Partial<PopperProps>;
+  PaperProps?: PaperProps;
+};
+export const CascadingMenuItem: FC<CascadingMenuItemProps> = ({
+  MenuItemComponent,
+  PopperProps,
+  PaperProps,
+  children,
+  ...props
+}) => {
+  const [anchorEl, setAnchorEl] = useState<HTMLElement | null>(null);
+
+  return (
+    <MenuItem
+      onMouseEnter={(evt) => setAnchorEl(evt.currentTarget)}
+      onMouseLeave={() => setAnchorEl(null)}
+      sx={{
+        // HACK: Prevents the menu item to be in active style state when the sub-menu is opened.
+        "&.MuiMenuItem-root": {
+          backgroundColor: Boolean(anchorEl)
+            ? "action.hover"
+            : "background.paper",
+        },
+        ...props?.sx,
+      }}
+      {...props}
+    >
+      {MenuItemComponent}
+      {Boolean(anchorEl) && (
```

Comment:
@agalin920 that's actually what I've been contemplating about as well recently and I just think it's easier to read if I cast it to Boolean instead of the !! operator. For example, checking if it's falsy will turn out to be `!Boolean(value)` instead of `!!!value`, I just find the first one to be more readable. But let me know if we should be using the !! instead so I can use that moving forward.